### PR TITLE
Classes

### PR DESCRIPTION
**Pull request by Fahim-Khan-P**
### **In this PR I:**

- [ ] Define a class called **Person** with instance variables `@id`, `@name`, and `@age`. Create methods `is_of_age?` and `can_use_services?` for this class. Additionally, include getters and setters methods for `@name` and `@age`, as well as a getter method for `@id`.
- [ ] Create a subclass of **Person** called **Student**. Include an instance variable `@classroom` and a method called `play_hooky` in this subclass.
- [ ] Create another subclass of **Person** called **Teacher**. This subclass should include an instance variable `@specialization`. Override the method `can_use_services?` from the **Person** superclass in this subclass.
- [ ] Follow **GitFlow**.
- [ ] Fix **Linter** errors.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class Person
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def initialize(age, name = 'Unkown', parents_permission: true)
```
Comment (franclobo) [inline person.rb]: - [ ] Similar to the default value in the name please 🙏🏽 use "=" to define the default value to `parent_permission` for the correct syntax.

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_service?
+    of_age? || @parents_permissiom
+  end
```
Comment (franclobo) [inline person.rb:22]: - [ ] Please 🙏🏽 move this method to the public section for being able to use by the other classes.

File: student.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Student < Person
+  def initialize(classroom, age, name = 'Unknown', parents_permission: true)
```
Comment (franclobo) [inline student.rb]: - [ ] Similar to the default value in the name please 🙏🏽 use "=" to define the default value to `parent_permission` for the correct syntax.

File: student.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    super(age, name, parents_permission: parents_permission)
```
Comment (franclobo) [inline student.rb]: - [ ] As you have done with the name it is not necessary to declare a default value to parent permission again, please 🙏🏽 remove the default value for the correct syntax.

File: teacher.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Teacher < Person
+  def initialize(specialization, age, name = 'Unknown', parents_permission: true)
```
Comment (franclobo) [inline teacher.rb]: - [ ] Similar to the default value in the name please 🙏🏽 use "=" to define the default value to `parent_permission` for the correct syntax.

File: teacher.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+    super(age, name, parents_permission: parents_permission)
```
Comment (franclobo) [inline teacher.rb]: - [ ] As you have done with the name it is not necessary to declare a default value to parent permission again, please 🙏🏽 remove the default value for the correct syntax.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi <STUDENT_MENTION>, 

Good job so far!

# ♻️ Required Changes ♻️

so-close-jim-halpert

You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- Good communication with code reviewer ✔️ 
- All classes are created 💯 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Student < Person
+  def initialize(classroom, age, name = 'Unknown', parents_permission = `true`)
```
Comment (youmari) [inline student.rb]: - [ ] Nice adding the required changes however you shouldn't put backticks around the true keyword that will return an empty string so please try to remove the backticks (single quotes )

File: teacher.rb [hunk 8]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Teacher < Person
+  def initialize(specialization, age, name = 'Unknown', parents_permission = `true`)
```
Comment (youmari) [inline teacher.rb]: - [ ] Nice adding the required changes however you shouldn't put backticks around the true keyword that will return an empty string so please try to remove the backticks (single quotes )

File: person.rb [hunk 9]
```diff
@@ -0,0 +1,23 @@
+class Person
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def initialize(age, name = 'Unkown', parents_permission = `true`)
```
Comment (youmari) [inline person.rb]: - [ ] Nice adding the required changes however you shouldn't put backticks around the true keyword that will return an empty string so please try to remove the backticks (single quotes )

**Review by youmari — CHANGES_REQUESTED** [review]
Hi @Fahim-Khan-P , 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
## Highlights 
✔️ No linters error
✔️ Adding required changes

## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by franclobo — APPROVED** [review]
Hi @Fahim-Khan-P, 

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct git flow ✔️ 

## Optional suggestions

- **[OPTIONAL]** Due to the parent_permission is a boolean you shouldn't use quotation marks, it should look like this 👉🏽  `def initialize(age, name = "Unknown", parent_permission = true)` please  🙏🏽 correct them in the corresponding lines of your code for avoiding troubles in the next milestone.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — COMMENTED** [review]
Hi @Fahim-Khan-P 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors 👍 
* Well written PR description 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**